src/pushsource/_impl/model/directory.py

```diff
@@ -0,0 +1,7 @@
+from .base import PushItem
+from .. import compat_attr as attr
+
+
+@attr.s()
+class DirectoryPushItem(PushItem):
+    """A :class:`~pushsource.PushItem` representing a directory."""
```

Comment:
Minor - this could do with another sentence or two of explanation, e.g. something like:

> On a directory push item, the `src` attribute contains the full path to a directory tree.
> It should generally be interpreted as a request to recursively publish that entire directory
> tree as-is.